pkg/logger/consolejson/json.go

```diff
@@ -0,0 +1,415 @@
+	// Event.WithDuration is rendered.
+	//
+	// Usually combined with also specifying the time unit. Setting this to true
+	// and the duration unit to time.Minute will result in showing a 30 seconds
+	// duration as 0.5 instead of rounding down to the integer 0.
+	//
+	// When set to false (which is the default) the duration is formatted as an
+	// integer.
+	TimeDurationUseFloat bool
+}
+
+// Default is a logger Sink that outputs JSON-formatted logs to the console
+// using its default settings.
+var Default = New(Config{})
+
+// New creates a new JSON-console logging Sink.
+func New(conf Config) logger.Sink {
+	conf.CallerFileField = prepareFieldName(conf.CallerFileField, "caller")
+	conf.CallerLineField = prepareFieldName(conf.CallerLineField, "line")
+	conf.ErrorField = prepareFieldName(conf.ErrorField, "error")
+	conf.LevelField = prepareFieldName(conf.LevelField, "level")
+	conf.MessageField = prepareFieldName(conf.MessageField, "message")
+	conf.ScopeField = prepareFieldName(conf.ScopeField, "scope")
+	conf.DateField = prepareFieldName(conf.DateField, "date")
+	return sink{&conf}
+}
+
+type sink struct {
+	config *Config
+}
+
+// NewContext creates a new JSON-console logging Context using the
+// same configuration as the one given when creating the Sink.
+func (s sink) NewContext() logger.Context {
+	return context{
+		Config: s.config,
+	}
+}
+
+type context struct {
+	*Config
+	fields     []byte
+	caller     string
+	callerLine int
+	scope      string
+}
+
+func (c context) WriteOut(level logger.Level, message string) {
+	var buf bytes.Buffer
+	buf.WriteString(`{"`)
+	buf.WriteString(c.LevelField)
+	buf.WriteString(`":"`)
+	buf.WriteString(levelString(level))
+	buf.WriteRune('"')
+
+	if !c.DisableDate {
+		buf.WriteString(`,"`)
+		buf.WriteString(c.DateField)
+		buf.WriteString(`":`)
+		writeEscapedString(&buf, time.Now().Format(time.RFC3339))
+	}
+
+	if !c.DisableCaller {
+		buf.WriteString(`,"`)
+		buf.WriteString(c.CallerFileField)
+		buf.WriteString(`":`)
+		writeEscapedString(&buf, c.caller)
+		if !c.DisableCallerLine {
+			buf.WriteString(`,"`)
+			buf.WriteString(c.CallerLineField)
+			buf.WriteString(`":`)
+			buf.WriteString(strconv.FormatInt(int64(c.callerLine), 10))
+		}
+	}
+
+	if c.scope != "" {
+		buf.WriteString(`,"`)
+		buf.WriteString(c.ScopeField)
+		buf.WriteString(`":`)
+		writeEscapedString(&buf, c.scope)
+	}
+
+	if message != "" {
+		buf.WriteString(`,"`)
+		buf.WriteString(c.MessageField)
+		buf.WriteString(`":`)
+		writeEscapedString(&buf, message)
+	}
+
+	buf.Write(c.fields)
+	buf.WriteString("}\n")
+
+	buf.WriteTo(os.Stdout)
+}
+
+func (c context) SetScope(value string) logger.Context {
+	c.scope = value
+	return c
+}
+
+func (c context) SetCaller(file string, line int) logger.Context {
+	c.caller, c.callerLine = file, line
+	return c
+}
+
+func (c context) SetError(value error) logger.Context {
+	c = c.appendRawFieldName(c.ErrorField)
+	c.fields = appendEscapedString(c.fields, value.Error())
+	return c
+}
```

Comment:
Implementation here doesn't seem to match the documentation for SetError

Reply:
Nice catch. I've refactored it a bunch.,